yum-plugin/src/com/tw/go/plugin/material/artifactrepository/yum/exec/HttpConnectionChecker.java

```diff
@@ -1,59 +1,75 @@
-import org.apache.commons.httpclient.methods.GetMethod;
-import org.apache.commons.httpclient.params.HttpConnectionParams;
+import org.apache.http.HttpStatus;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.*;
 
 import java.io.IOException;
 
 public class HttpConnectionChecker implements ConnectionChecker {
 
     public void checkConnection(String url, Credentials credentials) {
-        HttpClient client = getHttpClient();
-        if (credentials.isComplete()) {
-            org.apache.commons.httpclient.Credentials usernamePasswordCredentials = new UsernamePasswordCredentials(credentials.getUser(), credentials.getPassword());
-            client.getState().setCredentials(AuthScope.ANY, usernamePasswordCredentials);
-        }
-        GetMethod method = getGetMethod(url);
-        method.setFollowRedirects(false);
-        try {
-            int returnCode = client.executeMethod(method);
-            if (returnCode != HttpStatus.SC_OK) {
-                throw new RuntimeException(method.getStatusLine().toString());
+        try (CloseableHttpClient client = getHttpClient(credentials)) {
+            HttpGet method = getGetMethod(url);
+            try (CloseableHttpResponse response = client.execute(method)) {
+                if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+                    throw new RuntimeException(response.getStatusLine().toString());
+                }
             }
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
-    GetMethod getGetMethod(String url) {
-        return new GetMethod(url);
+    HttpGet getGetMethod(String url) {
+        return new HttpGet(url);
     }
 
-    HttpClient getHttpClient() {
-        HttpClient client = new HttpClient();
-        client.getParams().setIntParameter(HttpConnectionParams.CONNECTION_TIMEOUT, getSystemProperty("yum.repo.connection.timeout", 10 * 1000));
-        client.getParams().setSoTimeout(getSystemProperty("yum.repo.socket.timeout", 5 * 60 * 1000));
-        return client;
+    CloseableHttpClient getHttpClient(Credentials credentials) {
+        BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+
+        if (credentials.isComplete()) {
+            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(credentials.getUser(), credentials.getPassword()));
+        }
+
+        RequestConfig requestConfig = RequestConfig.custom().
+                setConnectTimeout(getSystemProperty("yum.repo.connection.timeout", 10 * 1000)).
+                setSocketTimeout(getSystemProperty("yum.repo.socket.timeout", 5 * 60 * 1000)).
+                setAuthenticationEnabled(true).
```

Comment:
Yes, I saw that. Didn't know how "mock" the mock server was, and whether it would do the right thing and emulate that behavior. :) As long as it's handled, that's fine.